### PR TITLE
cmake: Move cxx11 flags for fujitsu into the flag_handler method

### PR DESCRIPTION
**Pull request by chuckatkins**
This is a follow-up to #11839, which I merged prematurely.

See https://github.com/spack/spack/pull/11839#issuecomment-512863141

### CODE REVIEW

**Comment by chuckatkins** [comment]
@t-karatsu Can you verify this still works the way you need it to with the Fujitsu compilers? (it's one of the very few I don't have an active testing environment I can access).

**Comment by t-karatsu** [comment]
Thank you for submitting a new pull request. Please correct only one point.

Line 114: `self.compiler.name == 'fc'`
Fujitsu compiler's name in Spack is `fj`, so please correct as follows.
`self.compiler.name == 'fj'`

I confirmed to be able to install cmake with this fix!

File: var/spack/repos/builtin/packages/cmake/package.py [hunk 1]
```diff
@@ -110,11 +110,10 @@ class Cmake(Package):
...
     phases = ['bootstrap', 'build', 'install']
 
-    def setup_environment(self, spack_env, run_env):
-        if self.compiler.name == 'fj' \
-                and self.compiler.cxx11_flag \
-                not in self.spec.compiler_flags['cxxflags']:
-            spack_env.append_flags('CXXFLAGS', self.compiler.cxx11_flag)
+    def flag_handler(self, name, flags):
+        if name == 'cxxflags' and self.compiler.name == 'fc':
+            flags.append(self.compiler.cxx11_flag)
```
Comment (tgamblin) [inline var/spack/repos/builtin/packages/cmake/package.py]: What if the user tried to build with other possible cxx flags?  Is it going to raise an error when it sees `cxx14_flag` and `cxx11_flag` in there?  Also what if `cxx11_flag` is already in flags?

Comment (becker33) [reply var/spack/repos/builtin/packages/cmake/package.py]: @tgamblin @t-karatsu @chuckatkins I've added a commit to do some error checking in the flag handler. I've designed it to be easy to extend when future standards flags are added to the fj compiler going forward. Let me know what you think

Comment (chuckatkins) [reply var/spack/repos/builtin/packages/cmake/package.py]: lgtm but given that i jumped the gun on the merge last time I'll leave it to someone else to give it the green light.

Comment (t-karatsu) [reply var/spack/repos/builtin/packages/cmake/package.py]: I think that the correction is good and easy to understand. Thank you for making it better.

**Comment by t-karatsu** [comment]
I'm sorry, I did not check the operation when I commented before. I noticed that there is mistake of flag's name, so I fixed it on this pull request. Please confirm this pullreq?
https://github.com/spack/spack/pull/12114

**Pull request merged**